src/components/MultiHopTrade/components/TradeInput/components/TradeQuotes/TradeQuotes.tsx

```diff
@@ -71,29 +73,59 @@ export const TradeQuotes: React.FC<TradeQuotesProps> = memo(
     const quotes = useMemo(
       () =>
         sortedQuotes.map((quoteData, i) => {
-          const { index } = quoteData
+          const { swapperName, quote } = quoteData
 
-          // TODO(woodenfurniture): use quote ID when we want to support multiple quotes per swapper
-          const isActive = activeQuoteIndex === index
+          const isActive = activeQuoteId !== undefined && activeQuoteId.quoteId === quote?.id
           const bestQuoteSteps = bestQuoteData?.quote?.steps
           const lastStep = bestQuoteSteps?.[bestQuoteSteps.length - 1]
 
           return (
             <TradeQuote
               isActive={isActive}
-              isLoading={isLoading}
+              isLoading={isLoading || isSwapperFetching[swapperName]}
               isBest={i === 0}
-              key={index}
+              key={quote?.id ?? i}
               quoteData={quoteData}
               bestBuyAmountBeforeFeesCryptoBaseUnit={
                 lastStep?.buyAmountBeforeFeesCryptoBaseUnit ?? '0'
               }
             />
           )
         }),
-      [activeQuoteIndex, bestQuoteData?.quote?.steps, isLoading, sortedQuotes],
+      [activeQuoteId, bestQuoteData?.quote?.steps, isLoading, isSwapperFetching, sortedQuotes],
     )
 
+    // add some loading state per swapper so missing quotes have obvious explanation as to why they arent in the list
+    // only show these placeholders when quotes aren't already visible in the list
+    const fetchingSwappers = useMemo(() => {
+      return Object.entries(isSwapperFetching)
+        .filter(([_swapperName, isFetching]) => isFetching)
+        .filter(
+          ([swapperName, _isFetching]) =>
+            !sortedQuotes.some(quoteData => quoteData.swapperName === swapperName),
+        )
+        .map(([swapperName, _isFetching]) => {
+          const quoteData = {
+            quote: undefined,
+            swapperName: swapperName as SwapperName,
+            inputOutputRatio: 0,
+            errors: [],
+            warnings: [],
+          }
+          return (
+            <TradeQuote
+              isActive={false}
+              isLoading={true}
+              isBest={false}
+              key={`${swapperName}-fetching`}
+              // eslint-disable-next-line react-memo/require-usememo
```

Comment:
Can we add a comment on why we are doing this?

Reply:
This looks odd but it's correct and a bug with this lint rule, we can't `useMemo` here since we're already in the context of a memoized constant (the scope of which `quoteData` lives) and the lint rule is a bit derp in this scenario and fails to detect it

Reply:
here ya go a4f18f789871b23eac2ff7d2708614c0b33a8d4f